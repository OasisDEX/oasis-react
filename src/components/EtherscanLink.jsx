import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

export const ETHERSCAN_LINK_TYPE_ADDRESS = 'ETHERSCAN_LINK_TYPE_ADDRESS';
export const ETHERSCAN_LINK_TYPE_TRANSACTION = 'ETHERSCAN_LINK_TYPE_TRANSACTION';

const propTypes = PropTypes && {
  label: PropTypes.string,
  networkName: PropTypes.string,
  txHash: PropTypes.string,
  address: PropTypes.string,
  type: PropTypes.oneOf(
    [
      ETHERSCAN_LINK_TYPE_ADDRESS,
      ETHERSCAN_LINK_TYPE_TRANSACTION
    ]
  )
};
const defaultProps = {
  networkName: 'kovan'
};




const linkStyle = { maxWidth:'100px', textOverflow: 'ellipsis', display: 'inline-block', overflow: 'hidden' };

class EtherscanLink extends PureComponent {
  render() {
    const { txHash, label, networkName, address } = this.props;
    switch (this.props.type) {
      case ETHERSCAN_LINK_TYPE_TRANSACTION: {
        const url = `https://${networkName}.etherscan.io/tx/${txHash}`;
        return (
          <a style={linkStyle} href={url}>
            <span>{label||txHash}</span>
          </a>
        );
      }
      case ETHERSCAN_LINK_TYPE_ADDRESS: {
        const url = `https://${networkName}.etherscan.io/address/${address}`;
        return (
          <a style={linkStyle} href={url}>
            <span>{label||address}</span>
          </a>
        );
      }
    }
  }
}

EtherscanLink.displayName = 'EthercanLink';
EtherscanLink.propTypes = propTypes;
EtherscanLink.defaultProps = defaultProps;
export default EtherscanLink;
