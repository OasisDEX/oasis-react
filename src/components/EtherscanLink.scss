@import "../styles/modules/breakpoints";
@import "../styles/modules/typography";

.link {
  @include ellipsis(150px);
  @media (max-width: $xxs) {
    @include ellipsis(70px);
  }
  font-weight: 600;
}
