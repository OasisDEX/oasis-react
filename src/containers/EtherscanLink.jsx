import React, { PureComponent } from 'react';
import { PropTypes } from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes';

import { connect } from 'react-redux';
import EtherscanLink, {
  ETHERSCAN_LINK_TYPE_ADDRESS,
  ETHERSCAN_LINK_TYPE_TRANSACTION,
} from '../components/EtherscanLink';
import network from '../store/selectors/network';

const propTypes = PropTypes && {
  label: PropTypes.string,
  networkName: PropTypes.string,
  txHash: PropTypes.string,
  address: PropTypes.string,
  type: PropTypes.oneOf(
    [
      ETHERSCAN_LINK_TYPE_ADDRESS,
      ETHERSCAN_LINK_TYPE_TRANSACTION
    ]
  )
};

export class EthercanLinkWrapper extends PureComponent {
  render() {
    return (
      <EtherscanLink {...this.props}/>
    );
  }
}

export function mapStateToProps(state) {
  return {
    networkName: network.activeNetworkName(state)
  };
}
export function mapDispatchToProps() {
  return {};
}

EthercanLinkWrapper.propTypes = propTypes;
EthercanLinkWrapper.displayName = 'EthercanLink';

export default connect(mapStateToProps, mapDispatchToProps)(EthercanLinkWrapper);
