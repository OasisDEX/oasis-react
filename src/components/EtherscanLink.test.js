/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import EtherscanLink from './EtherscanLink';


describe('(Component) EtherscanLink', () => {
  it('should render', () => {
    const props = {
      children: (
        <span>test</span>
      )
    };
    const wrapper = shallow(
      <EtherscanLink {...props}/>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
