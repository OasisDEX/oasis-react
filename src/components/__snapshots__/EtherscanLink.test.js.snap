// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EtherscanLink should render 1`] = `""`;
